be/tool/beck.c

```diff
@@ -1556,6 +1632,167 @@ static void genadd(uint64_t gen)
 	}
 }
 
+static int nv_scan_offset_init(uint64_t workers_nr,
+			       uint64_t partitions_nr)
+{
+	uint64_t  p;
+
+	M0_PRE(partitions_nr <= AO_NR);
+
+	m0_mutex_init(&off_info.oi_lock);
+	m0_mutex_lock(&off_info.oi_lock);
+	off_info.oi_workers_nr = workers_nr;
+	off_info.oi_partitions_nr = partitions_nr;
+
+	off_info.oi_offset = m0_alloc(sizeof(struct worker_off_info) *
+				      workers_nr);
+	if (off_info.oi_offset == NULL) {
+		m0_mutex_unlock(&off_info.oi_lock);
+		return M0_ERR(-ENOMEM);
+	}
+
+	memset(&off_info.oi_pinfo, 0, sizeof(struct part_info));
+	for (p = 0; p < off_info.oi_partitions_nr; p++)
+		m0_mutex_init(&off_info.oi_part_lock[p]);
+	m0_mutex_unlock(&off_info.oi_lock);
+	return 0;
+}
+
+static void nv_scan_offset_fini(void)
+{
+	uint64_t  p;
+
+	m0_mutex_lock(&off_info.oi_lock);
```

Comment:
Why we need lock here?

Reply:
same as above, may be redundant for now, but in was added to ensure if there was nv_scan_offset_update() in progress in multithreaded env it will progress only after completion.

Reply:
removed in commit #d09e8e3